x/mongo/driver/operation.go

```diff
@@ -304,6 +308,16 @@ func (op Operation) Execute(ctx context.Context, scratch []byte) error {
 		return err
 	}
 
+	// If no deadline is set on the passed-in context, and op.Timeout is set, honor op.Timeout
+	// in new context for operation execution.
+	if _, deadlineSet := ctx.Deadline(); !deadlineSet && op.Timeout != nil {
+		newCtx, cancelFunc := context.WithTimeout(ctx, *op.Timeout)
```

Comment:
Is `context.WithTimeout (ctx, 0)` a context with no timeout? If the answer is "no":

```suggestion
        if *op.Timeout != time.Duration(0) {
		newCtx, cancelFunc := context.WithTimeout(ctx, *op.Timeout)  
        }
```

Reply:
`context.WithTimeout(ctx, 0)` is a context that is immediately expired, so the answer is no. I added something similar to what you have above. In particular, I check if no deadline is set on the passed-in context, and `op.Timeout` is set _and_ non-zero before honoring `op.Timeout` in a new context for operation execution.